src/MagnumPlugins/BasisImporter/BasisImporter.h

```diff
@@ -0,0 +1,188 @@
+
+@snippet MagnumPluginsBasisImporter.cpp basis-target-format-suffix
+
+If you want to be able to change the target format dynamically, you may want to
+set the "format" configuration of the plugin:
+
+@snippet MagnumPluginsBasisImporter.cpp basis-target-format-config
+
+If you link to the plugin you may also use @ref BasisImporter::setTargetFormat().
+
+*/
+class MAGNUM_BASISIMPORTER_EXPORT BasisImporter: public AbstractImporter {
+public:
+    /** @brief Default constructor */
+    explicit BasisImporter();
+
+    /** @brief Plugin manager constructor */
+    explicit BasisImporter(PluginManager::AbstractManager& manager, const std::string& plugin);
+
+    /**
+     * @brief Set the importers target format
+     * @param format The format to transcode Basis files to
+     */
+    void setTargetFormat(BasisTranscodingType format);
+
+    /**
+     * @brief The importers target format
+     * @param format The format to transcode Basis files to
+     *
+     * See @ref basisimporter-target-format.
+     */
+    BasisTranscodingType targetFormat();
+
+    ~BasisImporter();
+
+private:
+    struct State;
+
+    MAGNUM_BASISIMPORTER_LOCAL Features doFeatures() const override;
+    MAGNUM_BASISIMPORTER_LOCAL bool doIsOpened() const override;
+    MAGNUM_BASISIMPORTER_LOCAL void doClose() override;
+    MAGNUM_BASISIMPORTER_LOCAL void doOpenData(Containers::ArrayView<const char> data) override;
+
+    MAGNUM_BASISIMPORTER_LOCAL UnsignedInt doImage2DCount() const override;
+    MAGNUM_BASISIMPORTER_LOCAL Containers::Optional<ImageData2D> doImage2D(UnsignedInt id) override;
+
+    Containers::Array<unsigned char> _in;
+    Containers::Pointer<State> _state;
+};
+
+}}
+
+#include <Corrade/Containers/StaticArray.h>
+
+namespace Corrade { namespace Utility {
+
+/* Configuration value implementation for BasisTranscodingType */
+template<> struct ConfigurationValue<Magnum::Trade::BasisTranscodingType> {
+
+    static std::string toString(
+        const Magnum::Trade::BasisTranscodingType& value,
+        ConfigurationValueFlags)
+    {
+        #define _c(value) case Magnum::Trade::BasisTranscodingType::value: return #value;
+        switch(value) {
+            _c(Etc1)
+            _c(Etc2)
+            _c(Bc1)
+            _c(Bc3)
+            _c(Bc4)
+            _c(Bc5)
+            _c(Bc7M6OpaqueOnly)
+            _c(Pvrtc1_4OpaqueOnly)
+        }
+        #undef _c
+
+        return "<invalid>";
+    }
+
+    static Magnum::Trade::BasisTranscodingType fromString(
+        const std::string& value, ConfigurationValueFlags)
+    {
+        const Containers::StaticArray<8, const char*> FormatNames{
+            "Etc1", "Etc2", "Bc1", "Bc3", "Bc4", "Bc5",
+            "Bc7M6OpaqueOnly", "Pvrtc1_4OpaqueOnly"
+        };
+
+        Magnum::Int i = 0;
+        for(const char* name : FormatNames)
+            if(name == value) return Magnum::Trade::BasisTranscodingType(i);
+
+        return Magnum::Trade::BasisTranscodingType(-1);
+    }
+};
+
+}}
```

Comment:
Since this is only used from the test or plugin implementation (and never by plugin users), instead of having the `ConfigurationValue` along with everything else that it needs always present in the public header, what about putting it into a separate internal header instead?

Reply:
`Configuration.h`? That would need to include BasisImporter.h for the enum, I suppose. Or is there a way around that?

Reply:
Yes, it would, but that's okay -- whoever needs the `ConfigurationValue` will need the importer as well to use the values, but not the other way around.

Reply:
It's no longer required in the tests, so I will move it into BasisImporter.cpp